### PR TITLE
Add Configuration Requests Table

### PR DESCRIPTION
**Pull request by toddburnside**
As per https://docs.google.com/document/d/1f6FsL-u9I79Qt47jdmWnf0hZnwt3WYlkdR0RXQVyzTU/edit?tab=t.0

### CODE REVIEW

**Comment by rpiaggio** [comment]
I can't access the document. Screenshot of explore?

File: explore/src/main/scala/explore/programs/ProgramConfigRequestsTile.scala [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+          rowColumn(StatusColumnId, _.request.status)
+            .setCell(c => stateIcon(c.value))
+            .setSize(80.toPx)
+            .sortable
+        )
+    .useMemoBy((props, _, _) => // Rows
+      (props.configRequests, props.obs4ConfigRequests, props.targets)
+    ): (_, _, _) =>
+      (requests, obsMap, targets) => requests.map((_, r) => Row(r, obsMap, targets)).toList
+    // TODO: Save state
```
Comment (toddburnside) [inline explore/src/main/scala/explore/programs/ProgramConfigRequestsTile.scala:215]: I have another table to add. When I do that, I'll save the state of both of those, as well as the validations table, so I only have to do the UserPrefs updates once. I'll also need to add another tile.

**Comment by toddburnside** [comment]
> I can't access the document. Screenshot of explore?

Mockup of the tables. I'll see if I can get a different link.

**Review by rpiaggio — APPROVED** [review]
LGTM

**Comment by toddburnside** [comment]
> > I can't access the document. Screenshot of explore?
> 
> Mockup of the tables. I'll see if I can get a different link.

Does this one work? https://docs.google.com/document/d/1f6FsL-u9I79Qt47jdmWnf0hZnwt3WYlkdR0RXQVyzTU/edit?usp=sharing

**Comment by rpiaggio** [comment]
> > > I can't access the document. Screenshot of explore?
> > 
> > 
> > Mockup of the tables. I'll see if I can get a different link.
> 
> Does this one work? https://docs.google.com/document/d/1f6FsL-u9I79Qt47jdmWnf0hZnwt3WYlkdR0RXQVyzTU/edit?usp=sharing

No, but I'll see it in explore, don't worry.

**Pull request merged**